.github/workflows/packer-build-and-publish.yml

```diff
@@ -0,0 +1,223 @@
+          echo "job_workflow_ref=${jobWorkflowRefValue}" >> "${GITHUB_OUTPUT}"
+
+          repositoryNameValue=$(echo $GITHUB_REPOSITORY | cut -d'/' -f2)
+          echo "repository_name=${repositoryNameValue}" >> "${GITHUB_OUTPUT}"
+
+          getIdResponse=$(aws cognito-identity get-id --identity-pool-id "${COGNITO_IDENTITY_POOL_ID}" \
+          --account-id "${AWS_ACCOUNT_ID}" \
+          --logins '{"token.actions.githubusercontent.com":"'"${ACCESS_TOKEN}"'"}')
+          identityId=$(echo "${getIdResponse}" | jq -rc '.IdentityId')
+
+          cognitoIdentityTokenResponse=$(aws cognito-identity get-open-id-token --identity-id "${identityId}" \
+          --logins '{"token.actions.githubusercontent.com":"'"${ACCESS_TOKEN}"'"}')
+          cognitoIdentityOidcAccessToken=$(echo "${cognitoIdentityTokenResponse}" | jq -r '.Token')
+
+          echo "::add-mask::$cognitoIdentityOidcAccessToken"
+
+          awsCredentials=$(aws sts assume-role-with-web-identity \
+            --role-session-name "GitHubActions" \
+            --role-arn "${JUMP_ROLE_ARN}" \
+            --duration-seconds 18000 \
+            --web-identity-token "${cognitoIdentityOidcAccessToken}")
+
+          accessKeyId=$(echo "$awsCredentials" | jq -r ".Credentials.AccessKeyId")
+          echo "::add-mask::$accessKeyId"
+
+          secretAccessKey=$(echo "$awsCredentials" | jq -r ".Credentials.SecretAccessKey")
+          echo "::add-mask::$secretAccessKey"
+
+          sessionToken=$(echo "$awsCredentials" | jq -r ".Credentials.SessionToken")
+          echo "::add-mask::$sessionToken"
+
+          echo "AWS_ACCESS_KEY_ID=${accessKeyId}" >> "${GITHUB_ENV}"
+          echo "AWS_SECRET_ACCESS_KEY=${secretAccessKey}" >> "${GITHUB_ENV}"
+          echo "AWS_SESSION_TOKEN=${sessionToken}" >> "${GITHUB_ENV}"
+
+          expiration=$(echo "$awsCredentials" | jq -r ".Credentials.Expiration")
+
+          echo "Jump role session expires at: $expiration"
+
+      - name: Set up Packer
+        uses: hashicorp/setup-packer@1aa358be5cf73883762b302a3a03abd66e75b232 # v3.1.0
+
+      - name: Packer build
+        id: build
+        env:
+          GITHUB_REF: ${{ github.ref }}
+          GITHUB_JOB_WORKFLOW_REF: ${{ steps.auth.outputs.job_workflow_ref }}
+          GITHUB_REPOSITORY_NAME: ${{ steps.auth.outputs.repository_name }}
+        run: |
+          packer init ${{ matrix.template }}
+          packer build \
+            -var provider=aws \
+            -var aws_private_ami=true \
+            -var image_version="${{ github.ref_name }}" \
+            -var aws_assume_role_arn="arn:aws:iam::${{ matrix.account }}:role/github-actions/packer-role" \
+            -var aws_assume_role_session_name=GitHubActions \
+            ${{ matrix.template }}
+
+          echo "name=$(basename ${{ matrix.template }})" >> "${GITHUB_OUTPUT}"
+          echo "readme=$(git diff --name-only -- '*.md')" >> "${GITHUB_OUTPUT}"
```

Comment:
It shouldn't, a build for one template should only modify the readme for the relevant OS and architecture.

Since builds for each template are run in different jobs we will only have one modified readme for each.